index.js

```diff
@@ -20,39 +20,27 @@
 //. purpose: the [`typeof`][1] operator and [`Object.prototype.toString`][2].
 //. Each has pros and cons, but neither supports user-defined types.
 //.
-//. This package specifies an [algorithm][3] for deriving a _type identifier_
-//. from any JavaScript value, and exports an implementation of the algorithm.
-//. Authors of algebraic data types may follow this specification in order to
-//. make their data types compatible with the algorithm.
+//. This package exports a function which derives a _type identifier_ from any
+//. JavaScript value, and a [specification][3] for customizing the type
+//. identifier of a value.
 //.
-//. ### Algorithm
+//. ### Specification
 //.
-//. 1.  Take any JavaScript value `x`.
+//. For an algebraic data type to be compatible with the algorithm:
 //.
-//. 2.  If `x` is `null` or `undefined`, go to step 6.
-//.
-//. 3.  If `x.constructor` evaluates to `null` or `undefined`, go to step 6.
-//.
-//. 4.  If `x.constructor.prototype === x`, go to step 6. This check prevents a
-//.     prototype object from being considered a member of its associated type.
-//.
-//. 5.  If `typeof x.constructor['@@type']` evaluates to `'string'`, return
-//.     the value of `x.constructor['@@type']`.
-//.
-//. 6.  Return the [`Object.prototype.toString`][2] representation of `x`
-//.     without the leading `'[object '` and trailing `']'`.
-//.
-//. ### Compatibility
-//.
-//. For an algebraic data type to be compatible with the [algorithm][3]:
-//.
-//.   - every member of the type must have a `constructor` property pointing
+//.   - every member of the type MUST have a `constructor` property pointing
 //.     to an object known as the _type representative_;
 //.
-//.   - the type representative must have a `@@type` property; and
+//.   - the type representative MUST have a `@@type` property;
 //.
 //.   - the type representative's `@@type` property (the _type identifier_)
-//.     must be a string primitive, ideally `'<npm-package-name>/<type-name>'`.
+//.     MUST be a string primitive; and
+//.
+//.   - the `@@type` property SHOULD have format:
+//.     `'<namespace>/<name>@<version>'` - where `namespace` SHOULD equal the
+//.     NPM package name which defines the type, `name` SHOULD be the unique
+//.     name of the type, and `version` MAY be a numeric value representing the
+//.     version of the type.
```

Comment:
They are both optional because we want this version of the spec to be backwards compatible with the previous, which didn't enforce any kind of string format. It simply suggested prefixing the type with your package name. This is also the reason why I'm using SHOULD and not MUST.